### PR TITLE
MB-15684 replace MakeMTOShipment functions with BuildMTOShipment

### PR DESCRIPTION
**Pull request by StuTruss**
## Summary

Replace calls to MakeMTOShipment with BuildMTOShipment in all files except `pkg/testdatagen/*`

## Setup to Run Your Code

Run `make server_test` and make sure tests pass

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/factory/mto_shipment_factory.go [hunk 1]
```diff
@@ -87,9 +87,20 @@ func buildMTOShipmentWithBuildType(db *pop.Connection, customs []Customization,
...
+						Model: models.PostalCodeToGBLOC{
+							PostalCode: pickupAddress.PostalCode,
+							GBLOC:      "KKFA",
+						},
+					},
+				}, nil)
+			} else {
+				gbloc, err := models.FetchGBLOCForPostalCode(db, pickupAddress.PostalCode)
+				if gbloc.GBLOC == "" || err != nil {
+					FetchOrBuildPostalCodeToGBLOC(db, pickupAddress.PostalCode, "KKFA")
```
Comment (ahobson) [inline pkg/factory/mto_shipment_factory.go:102]: I don't believe you need to test with `models.FetchGLBOCForPostalCode` here first, it should be ok to always use the factory `FetchOrBuildPostalCodeToGBLOC`

File: pkg/handlers/ghcapi/mto_shipment_test.go [hunk 2]
```diff
@@ -55,34 +55,53 @@ func (suite *HandlerSuite) makeListMTOShipmentsSubtestData() (subtestData *listM
...
+	mtoShipment := factory.BuildMTOShipment(suite.DB(), []factory.Customization{
+		{
+			Model:    mto,
+			LinkOnly: true,
 		},
-	})
+		{
+			Model: models.MTOShipment{
+				Status:           models.MTOShipmentStatusApproved,
+				ShipmentType:     models.MTOShipmentTypeHHGIntoNTSDom,
```
Comment (ahobson) [inline pkg/handlers/ghcapi/mto_shipment_test.go:66]: It seems like the test now needs to specify `ShipmentType` and that is a change. Do you know why that would be?

File: pkg/services/mto_shipment/mto_shipment_creator_test.go [hunk 3]
```diff
@@ -68,10 +70,12 @@ func (suite *MTOShipmentServiceSuite) TestCreateMTOShipment() {
...
-		mtoShipment := testdatagen.MakeMTOShipment(appCtx.DB(), testdatagen.Assertions{
-			Move: subtestData.move,
-			Stub: true,
-		})
+		mtoShipment := factory.BuildMTOShipment(appCtx.DB(), []factory.Customization{
+			{
+				Model:    subtestData.move,
+				LinkOnly: true,
+			},
+		}, nil)
```
[the remark this answers is not in the document]
Comment (StuTruss) [reply pkg/services/mto_shipment/mto_shipment_creator_test.go:78]: that was me trying to test out a potential fix. when I set it to `nil` it still returned errors

File: pkg/handlers/ghcapi/mto_shipment_test.go (hunk 2 above)
Comment (StuTruss) [reply pkg/handlers/ghcapi/mto_shipment_test.go:66]: mainly for `StorageFacility` and `PickupAddress` based off the conditional statements made in `make_mto_shipment.go` those models shouldn't be populated under certain circumstances but somehow they still were getting merged in. With the new Build function that doesn't happen now unless certain shipmentTypes are passed in

**Pull request merged**